mass/core/optimal_filtering.py

```diff
@@ -261,12 +261,15 @@ class Filter:
     f_3db: Optional[float] = None
     cut_pre: int = 0
     cut_post: int = 0
-    _filter_type: str = ""
 
     @property
     def is_arrival_time_safe(self):
         """Is this an arrival-time-safe filter?"""
-        return self.convolution_lags == 1 and self.dt_values is not None
+        return False
+
+    @property
```

Comment:
Are we supposed to never make an object of type `Filter`? In that case consider naming is `BaseFilter` and `def __init__(self, ...): raise Error`? Move the error as early as possible.

Reply:
You're right. And...it turns out there's [a package](https://docs.python.org/3/library/abc.html) and a [special decorator](https://docs.python.org/3/library/abc.html#abc.abstractmethod) to enforce this. I'll use that.